client/components/carousel.jsx

```diff
@@ -1,53 +1,54 @@
-import React from 'react';
-
-// import React, { useState, useEffect } from 'react';
-// import jordanImage from '../../images/jordan.png';
-// import newBalanceImage from '../../images/new-balance.png';
-// import asicsImage from '../../images/asics.png';
-
-export default function Carousel({ handleAnchorClick, currentIndex, carouselImages, goToNext, goToPrevious }) {
-  // const [currentIndex, setCurrentIndex] = useState(0);
-
-  // const carouselImages = [
-  //   { url: jordanImage, title: 'Jordan' },
-  //   { url: newBalanceImage, title: 'New Balance' },
-  //   { url: asicsImage, title: 'Asics' }
-  // ];
-
-  // useEffect(() => {
-  //   const intervalId = setInterval(() => {
-  //     setCurrentIndex((currentIndex + 1) % carouselImages.length);
-  //   }, 5000);
-
-  //   return () => clearInterval(intervalId);
-  // });
-
-  // function goToPrevious() {
-  //   const isFirstSlide = currentIndex === 0;
-  //   const newIndex = isFirstSlide ? carouselImages.length - 1 : currentIndex - 1;
-  //   setCurrentIndex(newIndex);
-  // }
-
-  // function goToNext() {
-  //   const isLastSlide = currentIndex === carouselImages.length - 1;
-  //   const newIndex = isLastSlide ? 0 : currentIndex + 1;
-  //   setCurrentIndex(newIndex);
-  // }
+// import React from 'react';
+
+import React, { useState, useEffect } from 'react';
+import jordanImage from '../../images/jordan.png';
+import newBalanceImage from '../../images/new-balance.png';
+import asicsImage from '../../images/asics.png';
+
+// export default function Carousel({ handleAnchorClick, currentIndex, carouselImages, goToNext, goToPrevious }) {
+export default function Carousel({ handleAnchorClick }) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const carouselImages = [
+    { url: jordanImage, title: 'Jordan' },
+    { url: newBalanceImage, title: 'New Balance' },
+    { url: asicsImage, title: 'Asics' }
+  ];
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      setCurrentIndex((currentIndex + 1) % carouselImages.length);
+    }, 5000);
+
+    return () => clearInterval(intervalId);
+  });
```

Comment:
You probably want a dependency array here, otherwise this runs on _every_ render. It looks like you depend on `currentIndex` and `carouselImages.length`. Those should be in your dependency array.